src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -300,6 +300,15 @@ var ReactCompositeComponent = {
           'componentWillRecieveProps(). Did you mean componentWillReceiveProps()?',
         this.getName() || 'A component',
       );
+
+      if (isPureComponent(Component) && typeof inst.shouldComponentUpdate !== 'undefined') {
+        warning(
+          false,
```

Comment:
Just to be clear, does this mean if the app has two different pure components, the warning should only show up once? Or does it mean to only warn once for the same pure component used numerous times in different places?
I'm guessing it's the latter but asking just to be sure :)